src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -477,11 +477,13 @@ std::unique_ptr<InputMatcher> InputMatcher::Create(
 TriggeredPublisher::~TriggeredPublisher()
 {
   this->done = true;
+  this->srvDone = true;
   this->newMatchSignal.notify_one();
+  this->serviceMatchSignal.notify_one();
   if (this->workerThread.joinable())
-  {
     this->workerThread.join();
-  }
+  if (this->serviceWorkerThread.joinable())
+    this->serviceWorkerThread.join();
```

Comment:
I'm looking at other plugin files and we usually have braces for single-line if-statements. It's generally safer for future additions.
Applies for rest of file.